examples/pipeline_wavernn/datasets.py

```diff
@@ -0,0 +1,113 @@
+import os
+import random
+
+import torch
+import torchaudio
+from torch.utils.data.dataset import random_split
+from torchaudio.datasets import LJSPEECH
+
+from functional import label_to_waveform, mulaw_encode, waveform_to_label
+
+
+class MapMemoryCache(torch.utils.data.Dataset):
+    r"""Wrap a dataset so that, whenever a new item is returned, it is saved to memory.
+    """
+
+    def __init__(self, dataset):
+        self.dataset = dataset
+        self._cache = [None] * len(dataset)
+
+    def __getitem__(self, n):
+        if self._cache[n] is not None:
+            return self._cache[n]
+
+        item = self.dataset[n]
+        self._cache[n] = item
+
+        return item
+
+    def __len__(self):
+        return len(self.dataset)
+
+
+class Processed(torch.utils.data.Dataset):
+    def __init__(self, dataset, transforms):
+        self.dataset = dataset
+        self.transforms = transforms
+
+    def __getitem__(self, key):
+        item = self.dataset[key][0]
+        return self.process_datapoint(item)
+
+    def __len__(self):
+        return len(self.dataset)
+
+    def process_datapoint(self, waveform):
+        specgram = self.transforms(waveform)
+        return waveform.squeeze(0), specgram
+
+
+def gen_datasets_ljspeech(args, transforms):
```

Comment:
nit: how about `ljspeech_process_split` or `split_process_ljspeech` as a name?

Reply:
I prefer the second one. Updated.